src/database/operations/fiscalSponsorships/createOrUpdateFiscalSponsorship.ts

```diff
@@ -0,0 +1,28 @@
+import { generateCreateOrUpdateItemOperation } from '../generators';
+import { Writable } from '../../../types/Writable';
+import type { Id } from '../../../types';
+
+/**
+ * A type designed to deal with the backend DB implementation and our generator
+ * functions that probably does not need to be used elsewhere in the code. It
+ * has two Ids that will be validated as Ids before ever getting to this point.
+ * The type is only here and not exported because it is used twice here. If it is
+ * found to be useful elsewhere, hoist it out to our general `types` directory
+ * and export it from there following the usual pattern.
+ */
+interface FiscalSponsorship {
+	fiscalSponseeChangemakerId: Id;
```

Comment:
You probably considered this already but I could see `sponsee` being an unnecessary descriptor since we have `fiscalSponsor` as a descriptor of the other field.

In either case you also might consider removing `fiscal` in both places since the context of *fiscal* sponsorship is already established by the table name.

(these are both "thought" level feedback, please don't consider this a change request if you don't agree)

Reply:
I started there but changed it to include `fiscal` to be crystal clear in all contexts. Especially since I didn't expose a `FiscalSponsorship` type and instead pass the IDs as args, in that context it is nice to have the parameter names match the database columns and so forth. It's verbose but I still think it's better to keep it.